### PR TITLE
Issue 15209 - redundant error message on invalid field access

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=15209

While fixing [regression issue 14965](https://issues.dlang.org/show_bug.cgi?id=14965), I found this inconsistency.

### CODE REVIEW

File: src/mtype.d [hunk 1]
```diff
@@ -7979,23 +7979,6 @@ public:
...
-                checkAccess(e.loc, sc, e, d);
-                Expression b = new AddrExp(e.loc, e);
-                b.type = e.type.pointerTo();
-                b = new AddExp(e.loc, b, new IntegerExp(e.loc, v.offset, Type.tint32));
-                b.type = v.type.pointerTo();
-                b = new PtrExp(e.loc, b);
-                b.type = v.type.addMod(e.type.mod);
-                return b;
-            }
-        }
```
Comment (9rnsr) [inline src/mtype.d]: By removing this old code, `TypeStruct.dotExp` and `TypeClass.dotExp` will become consistent.

**Comment by 9rnsr** [comment]
This front-end cleanup is necessary for:

https://trello.com/c/ihKhGeiu/115-reg2-031-forward-reference-to-inferred-return-type-of-function-call-when-using-auto-return-type

**Comment by AndrejMitrovic** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**